src/Services/InventoryStatusService.php

```diff
@@ -0,0 +1,450 @@
+<?php
+
+/**
+ * @copyright 2020 Wayfair LLC - All rights reserved
+ */
+
+namespace Wayfair\Services;
+
+use Wayfair\Core\Contracts\LoggerContract;
+use Wayfair\Core\Helpers\AbstractConfigHelper;
+use Wayfair\Helpers\TranslationHelper;
+use Wayfair\Repositories\KeyValueRepository;
+
+/**
+ * Service for setting / getting state of Inventory Services
+ */
+class InventoryStatusService
+{
+  const OVERDUE_TIME_FULL = 90000;
+  const OVERDUE_TIME_PARTIAL = 1800;
+
+  const LOG_KEY_STATE_CHANGE = 'inventoryStateChange';
+  const LOG_KEY_STATE_CHECK = 'inventoryStateCheck';
+  const LOG_KEY_STATE_CLEAR = 'inventoryStateClear';
+  const LOG_KEY_INVENTORY_VERSION_CHANGE = 'inventoryVersionChange';
+
+  const RESPONSE_KEY_STATUS = 'status';
+  const RESPONSE_KEY_DETAILS = 'details';
+  const RESPONSE_DETAILS_KEY_LAST_COMPLETION_START = 'completedStart';
+  const RESPONSE_DETAILS_KEY_LAST_COMPLETION_END = 'completedEnd';
+  const RESPONSE_DETAILS_COMPLETED_AMOUNT = 'completedAmount';
+  const RESPONSE_DETAILS_KEY_LAST_ATTEMPT_TIMESTAMP = 'attemptedStart';
+  const RESPONSE_DETAILS_KEY_OVERDUE = 'overdue';
+
+  const STATE_IDLE = 'idle';
+  const FULL = 'full';
+  const PARTIAL = 'partial';
+
+  const DB_KEY_INVENTORY_DATA_VERSION = 'inventory_data_version';
+
+  const DB_KEY_INVENTORY_STATUS = 'inventory_status';
+
+  const DB_KEY_INVENTORY_LAST_COMPLETION_END_FULL = "full_inventory_last_completion_end";
+  const DB_KEY_INVENTORY_LAST_ATTEMPT_FULL = "full_inventory_last_attempt";
+  const DB_KEY_INVENTORY_LAST_COMPLETION_START_FULL = "full_inventory_last_completion_start";
+  const DB_KEY_INVENTORY_LAST_COMPLETION_AMOUNT_FULL = "full_inventory_last_completion_amount";
+
+  const DB_KEY_INVENTORY_LAST_COMPLETION_END_PARTIAL = "partial_inventory_last_completion";
+  const DB_KEY_INVENTORY_LAST_ATTEMPT_PARTIAL = "partial_inventory_last_attempt";
+  const DB_KEY_INVENTORY_LAST_COMPLETION_START_PARTIAL = "partial_inventory_last_completion_start";
+  const DB_KEY_INVENTORY_LAST_COMPLETION_AMOUNT_PARTIAL = "partial_inventory_last_completion_amount";
+
+  const DB_KEYS = [

```

Comment:
yeah we could rearrange this array to have the order of the constants above, if you think it would help. We just need to make sure that all of the constants are in here, so that the clean/flush operation knows what DB keys to clear out.

Reply:
This array has the order of the constants above. Devs should add to it when they add a new DB key. We don't want to make separate arrays, because that would mean more things to know (and loop over) when the clean/flush operation runs.